amazon-kinesis-client/src/main/java/software/amazon/kinesis/processor/ShardRecordProcessorFactory.java

```diff
@@ -25,4 +25,13 @@
      * @return
      */
     ShardRecordProcessor shardRecordProcessor();
+
+    /**
+     * Returns a new instance of the ShardRecordProcessor for a stream
+     * @param streamName
+     * @return ShardRecordProcessor
+     */
+    default ShardRecordProcessor shardRecordProcessor(String streamName) {
```

Comment:
stream name is ignored?

Reply:
Our code now internally uses shardRecordProcessor(String streamName) by default and falls back to shardRecordProcessor() if customer does not set the multistreamtracker. If they use mutistreamtracker, they need to override this method in their concrete impl to provide a stream specific record processor.